src/rules/flake8_executable/helpers.rs

```diff
@@ -12,6 +12,10 @@ pub enum ShebangDirective<'a> {
 }
 
 pub fn extract_shebang(line: &str) -> ShebangDirective {
+    // Minor optimization to avoid matches in the common case.
+    if !line.contains('!') {
+        return ShebangDirective::None;
```

Comment:
Good one

Reply:
I feel like my brain is too small to understand why this is so effective (it reduced time from ~120ms to 100ms). I guess we're exploiting the fact that we know `!` is very uncommon, which `regex` itself can't know.